api/src/wfl/module/covid.clj

```diff
@@ -652,52 +701,80 @@
   (update-method-configuration! executor reference)
   (firecloud/submit-method workspace methodConfiguration))
 
-(defn ^:private write-workflows!
-  "Write `workflows` to `details` table."
+(defn ^:private allocate-submission
+  "Write or allocate workflow records for `submission` in `details` table."
   [{:keys [details]                :as _executor}
    {:keys [referenceId]            :as _reference}
    {:keys [submissionId workflows] :as _submission}]
-  (jdbc/with-db-transaction [tx (postgres/wfl-db-config)]
-    (let [now     (OffsetDateTime/now)
-          to-rows (fn [{:keys [status workflowId] :as _workflow}]
-                    {:snapshot_reference_id referenceId
-                     :rawls_submission_id   submissionId
-                     :workflow_id           workflowId
-                     :workflow_status       status
-                     :updated               now})]
-      (jdbc/insert-multi! tx details (map to-rows workflows)))))
+  (letfn [(to-row [now {:keys [status workflowId] :as _workflow}]
+            {:reference  referenceId
+             :submission submissionId
+             :workflow   workflowId
+             :status     status
+             :updated    now})]
+    (jdbc/with-db-transaction [tx (postgres/wfl-db-config)]
+      (->> (map (partial to-row (utc-now)) workflows)
+           (jdbc/insert-multi! tx details)))))
+
+;; Workflows in newly created firecloud submissions may not have been scheduled
+;; yet and thus do not have a workflow uuid. Thus, we first "allocate" the
```

Comment:
much thus

Reply:
Big thanks for figuring this out.  I can see how I missed it when comparing strictly against the COVID demo script and mocking out external calls in the integration tests.  I want to think more about how I could have seen this sooner by examining and/or supplementing the Firecloud tests.